test/ami-existence-poll_test.js

```diff
@@ -0,0 +1,149 @@
+const {AmiExistencePoller} = require('../lib/ami-existence-poller');
+const sinon = require('sinon');
+const main = require('../lib/main');
+const assume = require('assume');
+
+describe('AMI Existence Poller', () => {
+  let state;
+  let sandbox = sinon.sandbox.create();
+  let defaultSR;
+
+  before(async () => {
```

Comment:
```javascript
 10   before(async() => {
 11     db = await main('state', {profile: 'test', process: 'test'});
 12     await db._runScript('drop-db.sql');
 13     await db._runScript('create-db.sql');
 14   });
```

I think the issues you're having with the table not existing is that the create-db.sql script isn't getting run before the tests run.  Since this file alphabetically sorts early, it probably is being run after the database is set up by another unit tests.  This highlights though, that the other DB tests should be dropping the database in their `after()` calls, to ensure consistent behaviour.